lib/assert.js

```diff
@@ -30,45 +30,49 @@ function test(ok, opts) {
 	}
 }
 
+function stringMsg(msg) {
+	return msg instanceof Error ? msg.message : typeof msg === "string" ? msg : msg ? JSON.stringify(msg) : null;
+}
+
 x.pass = msg => {
-	test(true, create(true, true, 'pass', msg, x.pass));
+	test(true, create(true, true, 'pass', stringMsg(msg), x.pass));
```

Comment:
Instead of wrapping every `msg`, you can just add the validation to the `create` function.